cms/static/js/views/settings/main.js

```diff
@@ -316,22 +349,43 @@ var DetailsView = ValidatingView.extend({
 
     uploadImage: function(event) {
         event.preventDefault();
+        var title = "", selector = "", image_key = "", image_path_key = "";
+        switch (event.currentTarget.id) {
+            case 'upload-course-image':
+                title = "Upload your course image.";
+                selector = "#course-image";
+                image_key = 'course_image_name';
+                image_path_key = 'course_image_asset_path';
+                break;
+            case 'upload-hero-image':
+                title = "Upload your hero image.";
+                selector = "#hero-image";
+                image_key = 'hero_image_name';
+                image_path_key = 'hero_image_asset_path';
+                break;
+            case 'upload-thumbnail-image':
+                title = "Upload your video thumbnail image.";
+                selector = "#thumbnail-image";
+                image_key = 'thumbnail_image_name';
+                image_path_key = 'thumbnail_image_asset_path';
+                break;
+        }
+
         var upload = new FileUploadModel({
-            title: gettext("Upload your course image."),
+            title: gettext(title),
```

Comment:
This will not work. You will need to call `gettext()` on each of the title declarations above.